### PR TITLE
nightlies: fix bazel compilation

### PR DESCRIPTION
**Pull request by otan**
Follow up from error possibly starting in #79808

Release note: None

### CODE REVIEW

**Comment by otan** [comment]
not sure if https://github.com/cockroachdb/cockroach/blob/e0e302f817c8dd8f22ff2fb497a5ead16be67a68/pkg/cmd/bazci/watch.go#L357 needs to be updated

File: build/teamcity/cockroach/nightlies/roachtest_compile_bits.sh [hunk 1]
```diff
@@ -16,7 +16,7 @@ chmod a+w bin/cockroach bin/roachtest bin/workload
 # Stage the geos libs in the appropriate spot.
 mkdir -p lib.docker_amd64
 chmod o+rwx lib.docker_amd64
-cp $BAZEL_BIN/c-deps/libgeos/lib/libgeos.so   lib.docker_amd64
-cp $BAZEL_BIN/c-deps/libgeos/lib/libgeos_c.so lib.docker_amd64
+cp $BAZEL_BIN/c-deps/libgeos_foreign/lib/libgeos.so   lib.docker_amd64
```
Comment (rickystewart) [inline build/teamcity/cockroach/nightlies/roachtest_compile_bits.sh:19]: I think you need to update the `bazel build` call at the beginning to add the flag `--config force_build_cdeps`.

**Review by rickystewart — APPROVED** [review]
Whoops, sorry I missed this.

**Comment by rickystewart** [comment]
> not sure if needs to be updated

No, `bazci` is fine. For context it copies the geos libraries out of where the build system generates them out into `bazel-bin/c-deps/libgeos` to streamline things.

**Comment by otan** [comment]
bors r=rickystewart

**Pull request merged**